airflow/jobs.py

```diff
@@ -2237,7 +2244,7 @@ def _process_backfill_task_instances(self,
                             session=session,
                             verbose=self.verbose):
                         ti.refresh_from_db(lock_for_update=True, session=session)
-                        if ti.state == State.SCHEDULED or ti.state == State.UP_FOR_RETRY:
+                        if ti.state in [State.SCHEDULED, State.UP_FOR_RETRY, State.UP_FOR_RESCHEDULE]:
```

Comment:
Lately I've being giving preference to `{}` instead of `[]`. Especially the set operations are faster when it comes to the `in`:
```
MacBook-Pro-van-Fokko:~ fokkodriesprong$ ipython
Python 3.6.6 (v3.6.6:4cf1f54eb7, Jun 26 2018, 19:50:54) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.0.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from airflow.utils.state import State

In [2]: task_states_list = [
   ...:     State.SUCCESS,
   ...:     State.RUNNING,
   ...:     State.FAILED,
   ...:     State.UPSTREAM_FAILED,
   ...:     State.SKIPPED,
   ...:     State.UP_FOR_RETRY,
   ...:     State.QUEUED,
   ...:     State.NONE,
   ...:     State.SCHEDULED,
   ...: ]

In [3]: task_states_set = {
   ...:     State.SUCCESS,
   ...:     State.RUNNING,
   ...:     State.FAILED,
   ...:     State.UPSTREAM_FAILED,
   ...:     State.SKIPPED,
   ...:     State.UP_FOR_RETRY,
   ...:     State.QUEUED,
   ...:     State.NONE,
   ...:     State.SCHEDULED,
   ...: }

In [4]: %timeit State.NONE in task_states_set
   ...: 
59.1 ns ± 2.07 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: %timeit State.NONE in task_states_list
176 ns ± 0.704 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
I agree that using a set may make sense here.

But just replacing the inline list with a set does probably not make sense because the set is created every time again, and I assume creating a set is a bit more expensive than creating a list because elements needs to be hashed, but probably it's negligible. So the set needs to be defined as a "constant" which would make the code harder to read.

An alternative we could use tuple `()`, which inline creation should be a slightly cheaper.

Reply:
Changed to tuple `()` because moving it to a constant really makes it less readable.